### DISCUSSION TITLE
curl GCC ... arm-none-eabi ARM Cortex-M0+ support??

### DISCUSSION THREAD

**Discussion by slugvision** [curl/curl#12256] [Q&A] [answered]
I would like to use curl for my c++ gnu cross compiler ..arm-none-eabi for ARM Cortex-M0+

Will it work?

Any hints for setting up CMAKE?

If not, will Tiny Curl work?

Best regards, Ole

**Comment by dfandrich** [comment, marked as answer]
curl compiles fine under ARM (with a C compiler—not C++), but the Cortex M0 processor are often found in devices with minimal flash and RAM, too little to run curl easily. Tiny curl helps get the size of curl down even more, but it doesn't help with everything else you need to run curl (TCP/IP stack, DHCP daemon, DNS resolver, network drivers, etc.)  See the INSTALL.md file for hints on reducing size. See also #12143 for more on this topic.

**Reply by slugvision** [reply]
Thanks a lot for your hints!
